### PR TITLE
+doc Add examples to Sink.actorRefWithAck

### PR DESCRIPTION
**Pull request by ktoso**
Addresses half of https://github.com/akka/akka/issues/22874

### CODE REVIEW

File: akka-docs/src/main/paradox/stream/stream-integrations.md [hunk 1]
```diff
@@ -68,6 +68,27 @@ If the target actor terminates the stream will be cancelled. When the stream is
...
+
+Scala
+:   @@snip [IntegrationDocSpec.scala]($code$/scala/docs/stream/IntegrationDocSpec.scala) { #actorRefWithAck-actor }
+
+Java
+:   @@snip [IntegrationDocTest.java]($code$/java/jdocs/stream/IntegrationDocTest.java) { #actorRefWithAck-actor }
+
+Note that replying to the sender of the elements (the "stream") is required as lack of those ack signals would be interpreted
+as back-pressure (as intended), and no new elements will be sent into the actor until it acknowledges some elements.
+Handling the other signals while is not required, however is a good practice, to see the state of the streams lifecycle
```
Comment (raboof) [inline akka-docs/src/main/paradox/stream/stream-integrations.md]: "While not required, it is good practice to handle the other signals as well: this way you can keep track of the stream lifecycle in the connected actor"?

File: akka-docs/src/main/paradox/stream/stream-integrations.md [hunk 2]
```diff
@@ -68,6 +68,27 @@ If the target actor terminates the stream will be cancelled. When the stream is
...
+:   @@snip [IntegrationDocSpec.scala]($code$/scala/docs/stream/IntegrationDocSpec.scala) { #actorRefWithAck-actor }
+
+Java
+:   @@snip [IntegrationDocTest.java]($code$/java/jdocs/stream/IntegrationDocTest.java) { #actorRefWithAck-actor }
+
+Note that replying to the sender of the elements (the "stream") is required as lack of those ack signals would be interpreted
+as back-pressure (as intended), and no new elements will be sent into the actor until it acknowledges some elements.
+Handling the other signals while is not required, however is a good practice, to see the state of the streams lifecycle 
+in the connected actor as well. Technically it is also possible to use multiple sinks targeting the same actor,
+however it is not common practice to do so, and one should rather investigate using a `Merge` stage for this purpose.
```
Comment (raboof) [inline akka-docs/src/main/paradox/stream/stream-integrations.md]: :+1:

File: akka-docs/src/test/java/jdocs/stream/IntegrationDocTest.java [hunk 3]
```diff
@@ -309,6 +310,41 @@ public Receive createReceive() {
   }
   //#ask-actor
 
+  //#actorRefWithAck-actor
+  static class Ack {}
```
Comment (raboof) [inline akka-docs/src/test/java/jdocs/stream/IntegrationDocTest.java]: Do we have a convention to wrap those in a `Messages` enclosing class or something similar? The `static` looks kind of weird now.

File: akka-docs/src/test/scala/docs/stream/IntegrationDocSpec.scala [hunk 4]
```diff
@@ -195,6 +197,62 @@ class IntegrationDocSpec extends AkkaSpec(IntegrationDocSpec.config) {
...
+  "actorRefWithAck" in {
+    //#actorRefWithAck
+    val words: Source[String, NotUsed] =
+      Source(List("hello", "hi"))
+
+    // sent from actor to stream to "ack" processing of given element
+    val AckMessage = AckingReceiver.Ack
+
+    // sent from stream to actor to indicate start, end or failure of stream:
+    val InitMessage = AckingReceiver.StreamInitialized
```
Comment (raboof) [inline akka-docs/src/test/scala/docs/stream/IntegrationDocSpec.scala]: I'd be OK with skipping those `val`s and adding the comments when passing the parameters

**Review by raboof — APPROVED** [review]

File: akka-docs/src/test/java/jdocs/stream/IntegrationDocTest.java [hunk 5]
```diff
@@ -325,6 +361,28 @@ public void askStage() throws Exception {
...
   }
 
+  @Test
+  public void actorRefWithAckExample() throws Exception {
+    //#actorRefWithAck
+    Source<String, NotUsed> words =
+      Source.from(Arrays.asList("hello", "hi"));
+
+    ActorRef receiver =
+        system.actorOf(Props(() -> new AckingReceiver()));
```
Comment (johanandren) [inline akka-docs/src/test/java/jdocs/stream/IntegrationDocTest.java]: `Props.create()`?

File: akka-docs/src/test/scala/docs/stream/IntegrationDocSpec.scala (hunk 4 above)
Comment (johanandren) [reply akka-docs/src/test/scala/docs/stream/IntegrationDocSpec.scala]: +1

**Review by johanandren — APPROVED** [review]
LGTM sans the compilation error.

**Comment by ktoso** [comment]
Will make this pass to include in release now

**Comment by ktoso** [comment]
Pushed fixed squashed now

File: akka-docs/src/test/java/jdocs/stream/IntegrationDocTest.java (hunk 3 above)
Comment (ktoso) [reply akka-docs/src/test/java/jdocs/stream/IntegrationDocTest.java]: Ouch sorry I have not addressed this feedback! Should I follow up with it you think?

**Pull request merged**